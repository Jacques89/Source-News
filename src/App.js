import React, { useState } from 'react'

import Navigation from './components/Navigation/Navigation'
import NewsContainer from './components/NewsContainer/NewsContainer'
import './App.scss'

const App = () => {
  return (
    <div>
      <Navigation />
      <NewsContainer />
    </div>
  )
}

export default App