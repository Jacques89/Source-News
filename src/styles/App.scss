body {
  background-color: rgb(155, 131, 100);
  background-color: rgb(9, 3, 3);
}
