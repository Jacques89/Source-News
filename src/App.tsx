import { useState, useEffect } from 'react'

import './styles/App.scss'

import Navigation from './features/Navigation/Navigation'
import NewsContainer from './pages/NewsPage/NewsPage'
import LoadingPage from './features/LoadingPage/LoadingPage'
import SearchInput from './features/SearchInput/SearchInput'
import Footer from './features/Footer/Footer'

require('dotenv').config()

type NewsProps = {
  category: string
  content: string
  img: string
  title: string
}

const App = () => {
  const [news, setNews] = useState<Array<NewsProps>>([])
  const [category, setCategory] = useState<string>('')
  const [loading, setLoading] = useState<boolean>(true)
  const [userInput, setUserInput] = useState<string>('')

  const newsAPI: string = `${process.env.REACT_APP_NEWS_API}`

  const dataFetch = async () => {
    try {
      // const request: Response = await fetch(newsAPI)
      // const response = await request.json()
      // setNews(response)
      setLoading(false)
      // return response
    } catch (error: unknown) {
      console.log('ERROR:', error)
    }
  }

  const changeCategory = (category: string) => {
    setCategory(category)
    setUserInput('')
  }
  const handleSubmit = (input: string) => {
    setUserInput(input)
  }

  useEffect(() => {
    dataFetch()
    setCategory('World')
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <div data-testid='app'>
      <Navigation changeCategory={changeCategory} category={category} />
      <SearchInput handleSubmit={handleSubmit} />
      {news.length > 1 ? (
        <NewsContainer news={news} category={category} userInput={userInput} />
      ) : null}
      {loading ? <LoadingPage /> : null}
      {news.length > 0 ? <Footer /> : null}
    </div>
  )
}

export default App
