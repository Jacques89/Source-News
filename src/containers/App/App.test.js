import React from 'react'
import App from './App'

import { render } from '@testing-library/react'

describe('App', () => {
  describe('when the fetch operation is pending', () => {
    it('should render', () => {
      const { getByTestId } = render(<App />)

      expect(getByTestId('app')).toBeTruthy()
    })
  })

  const mockNews = {
    news: [
      {
        category: 'World'
      },
      {
        category: 'Sports'
      }
    ]
  }

  describe('MockAPI', () => {
    const mockUrl = '/api/news'
    const dataFetch = jest.fn(url => mockNews)

    it('returns news from an api call', () => {
      expect(dataFetch(mockUrl)).toBe(mockNews)
      console.log(dataFetch)
    })
    
    it('called getNews with a mockUrl', () => {
      expect(dataFetch).toHaveBeenCalledWith(mockUrl)
    })
  })
})