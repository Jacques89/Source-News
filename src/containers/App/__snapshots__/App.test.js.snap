// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div
  data-testid="app"
>
  <div
    style="margin-top: -1000px; transition-duration: .8s;"
  >
    <header>
      <h1>
        Source News
      </h1>
      <h3>
        Straight from the sources mouth!
      </h3>
      <svg
        class="icon-menu"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        />
      </svg>
    </header>
    <nav
      style="opacity: 0; display: none; z-index: 1;"
    >
      <ul
        aria-label="news-categories"
      >
        <li
          aria-label="category"
          style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
        >
          World
        </li>
        <li
          aria-label="category"
          style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
        >
          Business
        </li>
        <li
          aria-label="category"
          style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
        >
          Sports
        </li>
        <li
          aria-label="category"
          style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
        >
          Entertainment
        </li>
        <li
          aria-label="category"
          style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
        >
          Technology
        </li>
      </ul>
    </nav>
  </div>
  <form
    action="#"
    class="search-bar"
    style="margin-top: -1000px; transition-duration: .8s;"
  >
    <button
      class="search-btn"
    >
      <svg
        class="search-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
    </button>
    <input
      placeholder="Search"
      type="text"
      value=""
    />
  </form>
</div>
`;
